src/Operation/CreateCollection.php

```diff
@@ -170,6 +174,10 @@ public function __construct($databaseName, $collectionName, array $options = [])
             unset($options['writeConcern']);
         }
 
+        if (isset($options['autoIndexId'])) {
```

Comment:
Should this also include a version check? As is, won't this trigger an error even if a user is using version 2.6 or 3.0?

Reply:
I considered that, and/or making an exception for the `local` database, which MongoDB 4.0 should [still allow](https://jira.mongodb.org/browse/SERVER-8237?focusedCommentId=1808020&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1808020).

I'll ask this question in [DRIVERS-473](https://jira.mongodb.org/browse/DRIVERS-473).

Making an exception for the `local` database would be easy, although restrictions based on server version would require us to move this notice to `execute()`, which is a bit lousy.

Reply:
> As is, won't this trigger an error even if a user is using version 2.6 or 3.0?

Thinking about this some more, I think we're right to raise this for all server versions. The documentation for this option has always stated:

> For replica sets, do not set ``autoIndexId`` to ``false``.

The logic behind the server not permitting this at all (outside of `local`) regardless of mongod type is that a standalone could later be promoted to a replica set. This is irrespective of server version, so I don't see the harm in annoying users with a notice in this case.

----

On a separate note, it may not even make sense to allow an exemption for `local`. The point of deprecating this is that we'd ultimately like to remove the option altogether in our next major version. Keeping it around for `local` would complicate that goal.

If we deprecate it for later removal, users that really want to use this option down the line can likely just run the `create` command manually.

Reply:
I think it makes sense that we don't allow an exemption for `local` and raise the error for all server versions.